assets/js/modules/analytics-4/datastore/api.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * `modules/analytics-4` data store: api.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+
+const baseInitialState = { };
+const baseActions = {};
+const baseControls = {};
+const baseReducer = ( state, { type } ) => {
+	switch ( type ) {
+		default: {
+			return state;
+		}
+	}
+};
+const baseResolvers = {};
+const baseSelectors = {
+	/**
+	 * Checks if the Admin API is working.
+	 *
+	 * Returns false if there are any errors for getProperties or getWebDataStreams.
+	 *
+	 * Returns true if getProperties or getWebDataStreams has fetched successfully.
+	 *
+	 * Else returns undefined.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(boolean | undefined)} True, false, or undefined.
+	 */
+	isAdminAPIWorking( state ) {
+		for ( const errorKey in state.errors ) {
+			if (
+				errorKey.startsWith( 'getProperties' ) ||
+				errorKey.startsWith( 'getWebDataStreams' )
+			) {
+				return false;
+			}
+		}
+
+		for ( const propertyList of Object.values( state.properties ) ) {
+			if ( propertyList && propertyList.length && propertyList.length > 0 ) {
+				for ( const webdatastreamList of Object.values( state.webdatastreams ) ) {
+					if ( webdatastreamList && webdatastreamList.length && webdatastreamList.length > 0 ) {
+						return true;
+					}
+				}
+			}
+		}
+
+		return undefined;
+	},
+};
+
+const store = Data.combineStores(
+	{
+		initialState: baseInitialState,
+		actions: baseActions,
+		controls: baseControls,
+		reducer: baseReducer,
+		resolvers: baseResolvers,
+		selectors: baseSelectors,
```

Comment:
Any of these values that are empty aren't technically needed; the `combineStores` function will accept `undefined` values just fine: https://github.com/google/site-kit-wp/blob/3fb4b73421597f329da0ce0022ba315a6cf18630/assets/js/googlesitekit/data/utils.js#L183-L201

I'd remove any of these that you aren't using, especially as it reduces boilerplate and saves creating the passthrough reducer in this file.

Reply:
makes sense @tofumatt 

I saw an empty `baseControls`  in the ga-4 properties datastores so I just copied that :smile: 

https://github.com/google/site-kit-wp/blob/666153eda0986002b2533c10f2821e0d8a3cc0f8/assets/js/modules/analytics-4/datastore/properties.js#L167

Removing now